CRM/Mailing/BAO/Mailing.php

```diff
@@ -1488,6 +1488,9 @@ private function getTokenData(&$token_a, $html = FALSE, &$contact, &$verp, &$url
         $url .= '"';
       }
       $data = $url;
+      if (empty($html)) {
+        $data = str_replace('&amp;', '&', $data);
```

Comment:
or this https://github.com/civicrm/civicrm-core/blob/master/CRM/Utils/String.php#L783 for that matter